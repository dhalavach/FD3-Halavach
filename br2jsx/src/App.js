import Br2jsx from './components/Br2jsx';
function App() {
  const str = 'первый<br>второй<br/>третий<br />последний';

  return <Br2jsx htmlString={str} />;
}

export default App;
