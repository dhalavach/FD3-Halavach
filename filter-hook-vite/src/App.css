#root {
  max-width: 1280px;
  margin: 0 auto;
  padding: 2rem;
  text-align: center;
}.App {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  height: 100vh;
  font-size: 16px;
}
.search-header {
  display: flex;
}
.search-text {
  margin-right: 10px;
  font-size: 16px;
  display: flex;
  align-items: center;
  justify-content: center;
}
#search-box {
  height: 20px;
}
#search-box[placeholder] {
  line-height: 20px;
  font-size: 20px;
}
#item-list {
  min-height: 260px;
}

.button-small{
margin-left: 10px;
}