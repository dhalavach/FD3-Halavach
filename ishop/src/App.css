table, th, td {
border: solid 2px black}
table {
width: 100%;
background-color: whitesmoke;
}
h1 {
text-align: center;
}