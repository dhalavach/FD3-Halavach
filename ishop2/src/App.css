table, th, td {
border: solid 2px black}
table {
width: 100%;
background-color: whitesmoke;
}
h1 {
text-align: center;
}

.select-color{
background-color: orange;
}
.primary-color {
background-color: whitesmoke;
}

