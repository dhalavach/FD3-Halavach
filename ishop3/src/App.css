table, th, td {
  border: solid 2px black}
  table {
  width: 100%;
  background-color: whitesmoke;
  }
  h1 {
  text-align: center;
  }
  
  .button-small{
    margin: 8px 8px;
    padding: 16px;
  }
  
  .select-color{
  background-color: orange;
  }
  .primary-color {
  background-color: whitesmoke;
  }
  
  img {
  display: inline-block;
  width: 20%;
  height: 20%;
  }

  .edit-form-fields-container {
  display: flex;
  flex-direction: column;
  max-width: 25%;
  margin-left: 0.5rem;
  }

  .edit-form-buttons {
  max-width: 25%;
  display: flex;
  flex-direction: row;
  justify-content: flex-start
  }
  
  