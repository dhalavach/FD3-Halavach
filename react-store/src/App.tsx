import './styles.css';
import data from './mockData.json';
import { useState } from 'react';
import Table from './components/Table';

function App() {
  const [products, setProducts] = useState(data.products);
  const [sort, setSort] = useState('');
  const [type, setType] = useState('');
  return (
    <div className='grid-container'>
      <header>
        <a href='/'>Cart</a>
      </header>
      <main>
        <div className='content'>
          <div className='main'>
            <Table products={products} />
          </div>

          <div className='sidebar'>Items</div>
        </div>
      </main>
      <footer>2023</footer>
    </div>
  );
}

export default App;
