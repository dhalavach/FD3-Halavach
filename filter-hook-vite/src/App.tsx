import Filter from './components/Filter';
import './App.css';
import mockData from './assets/mockData.json';

function App() {
  return (
    <div className='App'>
      <Filter data={mockData} />
    </div>
  );
}

export default App;
