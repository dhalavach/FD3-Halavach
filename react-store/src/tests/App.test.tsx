import App from "../App";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import "@testing-library/jest-dom";
import { BrowserRouter as Router } from "react-router-dom";
import { Provider } from "react-redux";
import store from "../components/Store";
import renderer from "react-test-renderer";

Object.defineProperty(window, "matchMedia", {
  writable: true,
  value: jest.fn().mockImplementation((query) => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(), // Deprecated
    removeListener: jest.fn(), // Deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })),
});

test("renders main page", () => {
  expect(
    render(
      <Provider store={store}>
        <Router>
          <App />
        </Router>
      </Provider>
    )
  ).toBeTruthy();
});

test("renders link to cart", () => {
  render(
    <Provider store={store}>
      <Router>
        <App />
      </Router>
    </Provider>
  );
  const cart = screen.getByText(/cart/i);
  expect(cart).toBeInTheDocument();
});

test("component matches the snapshot", () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Router>
          <App />
        </Router>
      </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
