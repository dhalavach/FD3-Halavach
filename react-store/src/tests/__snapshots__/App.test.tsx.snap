// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component matches the snapshot 1`] = `
<div
  className="wrapper"
>
  <div
    className="grid-container"
  >
    <header
      className="header"
    >
      <a
        className="app__link-about-tablet"
        href="/about"
        onClick={[Function]}
      >
        About
      </a>
      <div
        className="app__heading-tablet"
      >
        <h1>
          Computer Store
        </h1>
      </div>
      <a
        className="app__link-cart-tablet"
        href="/ordersList"
        onClick={[Function]}
      >
        Orders
      </a>
      <a
        className="app__link-cart-tablet"
        href="/cart"
        onClick={[Function]}
      >
        Cart
      </a>
    </header>
    <main>
      <div
        className="content"
      >
        <div
          className="main"
        >
          <div
            className="controls"
          >
            <div
              className="filter-order"
            >
              <div>
                <div
                  className="filter-sort"
                >
                  Order
                   
                  <select
                    onChange={[Function]}
                    value=""
                  >
                    <option
                      value=""
                    />
                    <option
                      value="az"
                    >
                      A-Z
                    </option>
                    <option
                      value="za"
                    >
                      Z-A
                    </option>
                    <option
                      value="ascending"
                    >
                      Price - ascending
                    </option>
                    <option
                      value="descending"
                    >
                      Price - descending
                    </option>
                  </select>
                </div>
                <div
                  className="filter-type"
                >
                  Filter
                  <select
                    onChange={[Function]}
                    value=""
                  >
                    <option
                      value=""
                    >
                      All
                    </option>
                    <option
                      value="motherboard"
                    >
                      Motherboard
                    </option>
                    <option
                      value="CPU"
                    >
                      CPU
                    </option>
                    <option
                      value="GPU"
                    >
                      GPU
                    </option>
                    <option
                      value="PSU"
                    >
                      PSU
                    </option>
                  </select>
                </div>
              </div>
            </div>
            <div
              className="search"
            >
              <div>
                Search
                 
                <input
                  data-testid="search__input-field"
                  id="search-box"
                  name="search"
                  onChange={[Function]}
                  placeholder="search products"
                  type="search"
                  value=""
                />
              </div>
            </div>
          </div>
          <div>
            <div>
              <span>
                Loading...
              </span>
              <div
                className="MuiBox-root css-1mdvpvv"
              />
              <nav
                aria-label="pagination navigation"
                className="MuiPagination-root MuiPagination-text css-1oj2twp-MuiPagination-root"
              >
                <ul
                  className="MuiPagination-ul css-wjh20t-MuiPagination-ul"
                >
                  <li>
                    <button
                      aria-label="Go to previous page"
                      className="MuiButtonBase-root Mui-disabled MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular Mui-disabled MuiPaginationItem-previousNext css-yuzg60-MuiButtonBase-root-MuiPaginationItem-root"
                      disabled={true}
                      onBlur={[Function]}
                      onClick={[Function]}
                      onContextMenu={[Function]}
                      onDragLeave={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchMove={[Function]}
                      onTouchStart={[Function]}
                      tabIndex={-1}
                      type="button"
                    >
                      <svg
                        aria-hidden={true}
                        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-g2z002-MuiSvgIcon-root-MuiPaginationItem-icon"
                        data-testid="NavigateBeforeIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                        />
                      </svg>
                    </button>
                  </li>
                  <li>
                    <button
                      aria-label="Go to next page"
                      className="MuiButtonBase-root Mui-disabled MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular Mui-disabled MuiPaginationItem-previousNext css-yuzg60-MuiButtonBase-root-MuiPaginationItem-root"
                      disabled={true}
                      onBlur={[Function]}
                      onClick={[Function]}
                      onContextMenu={[Function]}
                      onDragLeave={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchMove={[Function]}
                      onTouchStart={[Function]}
                      tabIndex={-1}
                      type="button"
                    >
                      <svg
                        aria-hidden={true}
                        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-g2z002-MuiSvgIcon-root-MuiPaginationItem-icon"
                        data-testid="NavigateNextIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                        />
                      </svg>
                    </button>
                  </li>
                </ul>
              </nav>
              <ul
                className="products"
              />
              <div />
            </div>
          </div>
        </div>
        <div
          className="sidebar hidden"
        >
          <div>
            <div
              className="cart cart-header"
            >
              Empty
            </div>
          </div>
        </div>
      </div>
    </main>
    <footer>
      2023
    </footer>
  </div>
</div>
`;
